### DISCUSSION TITLE
[v4] Security Vulnerabilities

### DISCUSSION THREAD

**Discussion by jcjp** [adonisjs/core#3613] [Help] [answered]
Hi Adonis Team,

Since its not 100% smooth when transitioning from `v4` to `v5` how do we fix security vulnerabilities?

- Is it okay to just update the libraries under your project to the latest version even on `v4`?
- Will `v4` have a LTS or will we be forced to migrate to `v5`?
- If there is already a very straight forward and easy way could I have the link? (I only know this one so far: https://docs.adonisjs.com/releases/out-of-preview)

Please close this if deemed invalid, and point me to the correct location thanks!

**Comment by RomainLanz** [comment, marked as answer]
Hey @jcjp! 👋🏻 

> Is it okay to just update the libraries under your project to the latest version even on v4?

No. You should always check for any breaking change on those libraries. If you are talking about `@adonisjs` library, most of them have been updated for V5 and therefore, they don't work on V4.

> Will v4 have a LTS or will we be forced to migrate to v5?

No. Version 5 have been released nearly two years ago, the version 4 has been discontinued.

> If there is already a very straight forward and easy way could I have the link? 

I would recommend scaffolding a new application and starting to copy-paste your code by modifying what needs to be modified. Depending on the application size, it shouldn't take too long to update.

I can also recommend [following @ammezie course](https://adonismastery.com/courses/migrating-to-adonisjs-5) on the subject.

**Reply by jcjp** [reply]
Thanks for the answer, where could I post some questions with regards to migrating v4 to v5 (e.g. How to translate v4 providers to v5)? Would it be okay to post here under the Github discussions or is there a preferred medium?

**Reply by RomainLanz** [reply]
Yeah, all good with GitHub Discussion or the Discord server!
